frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/utils.ts

```diff
@@ -43,6 +43,12 @@ export const migratePipelineRun = (pipelineRun: PipelineRun): PipelineRun => {
   return newPipelineRun;
 };
 
+export const getPipelineName = (pipeline?: Pipeline, latestRun?: PipelineRun) =>
+  pipeline
+    ? pipeline.metadata.name
+    : latestRun.spec.pipelineRef?.name ??
```

Comment:
if you don't have a pipeline or a latestRun, you'll get an NPE 🙂  -- when you write tests you will see the logic shift.

Reply:
We wouldn't reach the call to `getPipelineName` if pipeline and `latestRun` both are absent, since it is handled at the very beginning of the definition of `getPipelineRunData`

Reply:
This function knows nothing of where it was called. If you write fully comprehensive tests for this function, you'd pass null for both values and see it crashes. Please add more tests.

Reply:
added a check in the function to handle the scenario when both arguments are falsy